extensions/resource-deployment/src/interfaces.ts

```diff
@@ -190,6 +190,17 @@ export interface NotebookInfo {
 	linux: string;
 }
 
+export enum OsType {
```

Comment:
Using enums like this is generally not a good idea because you're going to end up with a flag-checking-hell essentially. It's a minor issue so feel free to ignore, but a lot of the places you're needing to check for this could be written with OOP where each OS gets its own implementation of a class to know what its doing.